16649 general contrast issues

andrewgormley opened the pull request:
<!--
    Thank you for your interest in contributing to NetBox! Please note that
    our contribution policy requires that a feature request or bug report be
    approved and assigned prior to opening a pull request. This helps avoid
    waste time and effort on a proposed change that we might not be able to
    accept.

    IF YOUR PULL REQUEST DOES NOT REFERENCE AN ISSUE WHICH HAS BEEN ASSIGNED
    TO YOU, IT WILL BE CLOSED AUTOMATICALLY.

    Please specify your assigned issue number on the line below.
-->
### Fixes: #16649

<!--
    Please include a summary of the proposed changes below.
-->

- [x]  (Footer) Date and version text low contrast [dark, global]
- [x]  (Header) Secondary text low contrast [dark, global]
- [ ]  (Racks) Low contrast labels [both, multiple]
Leaving labels off this task for now, per our last discussion about labels needing to stay the same colors.

- [x]  (Object) Low contrast on object title sub items eg. created 2020-10-10 [dark, multiple]
- [x]  (Misc) Low contrast on form error text [both, multiple] 
**Fixed for light. Dark mode is not possible with the current fontsize, and colors.**

- [x]  (Toast) Low contrast for warning toast [light, multiple] 
**Improved the WCAG contrast score but not quite to AA standard as orange on light bg are both classed high contrast colors according to WCAG but this is okay with the warning visual aids**

- [x]  (Misc) Low contrast on placeholder module placeholder text [dark, multiple]

Commit:
fixes #16647: navigation contrast issues updated

Commit:
fixes #16651: table contrast issues new

Commit:
fixed #16649: general contrast issues

andrewgormley commented:
This PR needs to be reviewed in a sequential order, please review https://github.com/netbox-community/netbox/pull/16750 and https://github.com/netbox-community/netbox/pull/16751 first.

jeremystretch commented on netbox/project-static/styles/overrides/_tabler.scss:
```diff
@@ -35,12 +35,85 @@ pre {
-  // Adjust table anchor link contrast as not enough contrast in dark mode
-  filter: brightness(110%);
+table {
+  a {
+    color: $dark-teal;
```
This can probably be removed as it appears to be redundant.

andrewgormley replied:
removed.

jeremystretch commented on netbox/project-static/styles/overrides/_tabler.scss:
```diff
@@ -35,12 +35,85 @@ pre {
   background-color: rgba(var(--tblr-primary-rgb),.48)
 }
+
+// Side navigation
+.navbar-vertical {
+  background-color: $rich-black;
+  .dropdown-item {
+    color: white!important;
+  }
+  .text-secondary {
+    color: $gray-400!important;
+  }
+}
```
Let's move these to `transitional/_navigation.scss` just to keep all these tweaks in one place.

andrewgormley replied:
moved as requested.

jeremystretch commented on netbox/project-static/styles/overrides/_tabler.scss:
```diff
@@ -35,12 +35,85 @@ pre {
+    .btn {
+      color: var(--#{$prefix}body-color);
+    }
+    th.orderable a {
+      color: var(--#{$prefix}body-color);
+    }
```
Redundant to lines 42-44?

andrewgormley replied:
Removed.

jeremystretch reviewed (changes requested):
Thanks @andrewgormley!

I have one minor concern regarding the submenu headings in the navigation menu: At first glance it feels difficult to distinguish the headings from the link items as they're very similar in color.
![screenshot](https://github.com/netbox-community/netbox/assets/13487278/ca891463-0ef6-495c-8f33-565a4416307e)

jeremystretch commented:
Just noticed another issue: The card headings are the same color as the page behind them:

![screenshot](https://github.com/netbox-community/netbox/assets/13487278/d14f8706-b244-4dee-926a-7052ad719ae7)

Commit:
fixes #16649: feedback changes

andrewgormley commented:
@jeremystretch thanks for the review, I've fixed the issues. Regarding the navigation sub headers, I can definitely see what you mean, I've changed them to a primary color and updated the border to match, let me know what you think.

I've also included another small change to update the footer icon colors which we forgot to include as a sub-task.

jeremystretch reviewed (approved):
Thanks @andrewgormley, looks good!
